moto/networkmanager/models.py

```diff
@@ -195,6 +360,186 @@ def describe_global_networks(
+            description=description,
+            location=location,
+            tags=tags,
+            account_id=self.account_id,
+            partition=self.partition,
+        )
+        site_id = site.site_id
+        self.sites[global_network_id][site_id] = site
+        return site
+
+    def delete_site(self, global_network_id: str, site_id: str) -> Site:
+        if global_network_id not in self.global_networks:
+            raise ResourceNotFound(global_network_id)
+        if global_network_id not in self.sites:
+            raise ResourceNotFound(site_id)
+        gn_sites = self.sites[global_network_id]
+        site = gn_sites.pop(site_id)
+        site.state = "DELETING"
+        return site
+
+    @paginate(pagination_model=PAGINATION_MODEL)
+    def get_sites(self, global_network_id: str, site_ids: List[str]) -> List[Site]:
+        gn_sites = self.sites.get(global_network_id) or {}
+        queried = []
+        if not site_ids:
+            queried = list(gn_sites.values())
+        elif isinstance(site_ids, List) and len(site_ids) == 1:
+            site_id = site_ids[0]
+            if site_id not in gn_sites:
+                raise ResourceNotFound(site_id)
+            queried.append(gn_sites[site_id])
+        else:
+            for id in site_ids:
+                if id in gn_sites:
+                    q = gn_sites[id]
+                    queried.append(q)
+
+        return queried
+
+    def create_link(
+        self,
+        global_network_id: str,
+        description: Optional[str],
+        type: Optional[str],
+        bandwidth: Dict[str, Any],
+        provider: Optional[str],
+        site_id: str,
+        tags: Optional[List[Dict[str, str]]],
+    ) -> Link:
+        # check if global network exists
+        if global_network_id not in self.global_networks:
+            raise ResourceNotFound(global_network_id)
+        link = Link(
+            global_network_id=global_network_id,
+            description=description,
+            type=type,
+            bandwidth=bandwidth,
+            provider=provider,
+            site_id=site_id,
+            tags=tags,
+            account_id=self.account_id,
+            partition=self.partition,
+        )
+        self.links[global_network_id][link.link_id] = link
+        return link
+
+    @paginate(pagination_model=PAGINATION_MODEL)
+    def get_links(
+        self,
+        global_network_id: str,
+        link_ids: List[str],
+        site_id: str,
+        type: str,
+        provider: str,
+    ) -> List[Link]:
+        # TODO: Implement filtering by site_id, type, provider
+        gn_links = self.links.get(global_network_id) or {}
+        queried = []
+        if not link_ids:
+            queried = list(gn_links.values())
+        elif isinstance(link_ids, List) and len(link_ids) == 1:
+            link_id = link_ids[0]
+            if link_id not in gn_links:
+                raise ResourceNotFound(link_id)
```

Comment:
Is this error only raised when a single `link_id` is supplied? I'd would expect the logic to be the same regardless of how many `link_ids` are supplied.

There are three possible logic paths here of course, I don't know what AWS does:
 - error only if none of the provided `link_ids` are correct
 - error if a single link is found, and show that ID
 - error when some links are not found, and supply all `link_ids` that are not found

Reply:
Tested it, it'll return an empty list. No error. Fixing logic